### ISSUE TITLE
Inform the front-end about the total number of logical lines

### ISSUE THREAD

**Issue by eyelash** [google/xi-editor#1067]
Right now the front-end has no way of knowing the total number of logical lines. This information is needed for example to compute the correct gutter width.

**Comment by cmyr** [comment]
Yep good call. I'll add this along with 'width of the widest line' as the next batch of this work.
